filesystem/azure.go

```diff
@@ -0,0 +1,195 @@
+	}
+	containerName, blobName, err := convertToAzurePath(name)
+	if err != nil {
+		return nil, err
+	}
+	if !exist {
+		_, err := c.blobClient.CreateContainerIfNotExists(containerName, storage.ContainerAccessTypeBlob)
+		if err != nil {
+			return nil, err
+		}
+		err = c.blobClient.CreateBlockBlob(containerName, blobName)
+		if err != nil {
+			return nil, err
+		}
+	}
+	return &AzureFile{
+		path  : name,
+		logger : log.New(os.Stdout, "", log.Lshortfile|log.LstdFlags),
+		client : c.blobClient,
+	}, nil
+}
+
+func (f *AzureFile) Write(b []byte) (int, error) {
+	cnt, blob, err := convertToAzurePath(f.path)
+	if err != nil {
+		return 0, nil
+	}
+	blockList, err := f.client.GetBlockList(cnt, blob, storage.BlockListTypeAll)
+	blocksLen := len(blockList.CommittedBlocks) + len(blockList.UncommittedBlocks)
+	blockId := base64.StdEncoding.EncodeToString([]byte(fmt.Sprintf("%011d\n", blocksLen - 1)))
+	err = f.client.PutBlock(cnt, blob, blockId, b)
+	blockList, err = f.client.GetBlockList(cnt, blob, storage.BlockListTypeAll)
+	amendList := []storage.Block{}
+	for _, v := range blockList.CommittedBlocks {
+		amendList = append(amendList, storage.Block{v.Name, storage.BlockStatusCommitted})	
+	}
+	for _, v := range blockList.UncommittedBlocks {
+		amendList = append(amendList, storage.Block{v.Name, storage.BlockStatusUncommitted})
+	}
+	err = f.client.PutBlockList(cnt, blob, amendList)
+	if (err != nil) {
+		return 0, nil
+	}
+	return 0, nil
+}
+
+func (f *AzureFile) Close() error {
+	return nil
+}
+
+// AzureClient -> Glob function
+// Syntax : Adf{1,3}?ee*/ytsd.*
+// Follow regexp syntax except "/"
+func (c *AzureClient) Glob(pattern string) (matches []string, err error) {
+	afterSplit := strings.Split(pattern, "/")
+	cntPattern, blobPattern := afterSplit[0], afterSplit[1]
+	if len(afterSplit) != 2 {	
+		return nil, fmt.Errorf("Glob pattern should follow the Syntax")
+	}
+	resp, err := c.blobClient.ListContainers(storage.ListContainersParameters{Prefix: ""})
+	if err != nil {
+		return nil, err
+	}
+	for _, cnt := range resp.Containers {
+		if match, err := regexp.MatchString(cntPattern, cnt.Name); match && err == nil {
+			resp, err := c.blobClient.ListBlobs(cnt.Name, storage.ListBlobsParameters{
+				Marker:     ""})
+			if err != nil {
+				return nil, err
+			}
+			for _, v := range resp.Blobs {
+				if match, err := regexp.MatchString(blobPattern, v.Name); match && err == nil {
+					matches = append(matches, cnt.Name + "/" + v.Name)
+				}
+			}
+		}
+	}
+	return matches, nil
+}
+
+//NewAzureClient function
+// NewClient constructs a StorageClient and blobStorageClinet. 
+// This should be used if the caller wants to specify 
+// whether to use HTTPS, a specific REST API version or a
+// custom storage endpoint than Azure Public Cloud.
+// Recommended API version "2014-02-14"
+// synax : 
+// "spluto", "b7yy+C33a//uLE62Og9CkKDHRLNErMrbX40nKUxiTimgOvkP3MhEbjObmRxumda9grCwY8zqL6nLNcKCAS40Iw==", "core.chinacloudapi.cn", "2014-02-14", true
```

Comment:
Removing these leaked data might be trickier than you think (https://help.github.com/articles/remove-sensitive-data/). If you have revoked the key, the I think it's  fine to let it be.